### PR TITLE
devicestate: ensure permissions of /var/lib/snapd/void are correct

### PR DESCRIPTION
**Pull request by mvo5**
During the testflinger testing I saw some failures in the
`tests/main/cwd` test where the permissions of the
`/var/lib/snapd/void` directory are incorrect. On systems that
use packaging tools we force the permissions to be 0111 on each
refresh (see `debian/snapd.postinst`). However no such mechanism
exists on core devices. So if for some reason the permissions get
corrupted they never get fixed.

This commit adds code that ensures the permissions are fixed
automatically here too.

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -183,9 +183,26 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 	boot.RunFDESetupHook = m.runFDESetupHook
 	hookManager.Register(regexp.MustCompile("^fde-setup$"), newFdeSetupHandler)
 
+	// ensure permissions are ok
+	if err := ensureFileDirPermissions(); err != nil {
```
Comment (bboozzoo) [inline overlord/devicestate/devicemgr.go:187]: Hm here, or in StartUp(). As a precedent, we already have some code that sets up /var/lib/snapd/save mount in startup. Also StartUp() runs some early config code thus mutating the device state, so perhaps it'd be better to move the void fixup there too, such that Manager() is mostly about reading modeenv and registering various hooks.

**Review by stolowski — COMMENTED** [review]
LGTM overall, but one nitpick re the comment (and @bboozzoo 's suggestion sounds good).

File: overlord/devicestate/devicemgr.go (hunk 1 above)
Comment (mvo5) [reply overlord/devicestate/devicemgr.go]: Updated

**Review by bboozzoo — APPROVED** [review]
LGTM

File: overlord/devicestate/devicemgr.go [hunk 2]
```diff
@@ -183,9 +183,26 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
...
+		logger.Noticef("%v", fmt.Errorf("cannot ensure device file/dir permissions: %v", err))
+	}
+
 	return m, nil
 }
 
+func ensureFileDirPermissions() error {
+	// ensure the void dir has correct permissions
```
[the remark this answers is not in the document]
Comment (mvo5) [reply overlord/devicestate/devicemgr.go]: Thanks, added a bit more context now.

**Pull request merged**